tmc-langs-r/src/test/java/fi/helsinki/cs/tmc/langs/r/RPluginTest.java

```
@@ -0,0 +1,203 @@
+        assertArrayEquals(points, result.points.toArray());
+    }
+
+    @Test
+    public void testGetTestCommand() {
+        String[] command = new String[]{"Rscript"};
+        String[] args;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            args = new String[]{"-e", "\"library('tmcRtestrunner');run_tests()\""};
+        } else {
+            args = new String[]{"-e", "library(tmcRtestrunner);run_tests()"};
+        }
+        String[] expectedCommand = ArrayUtils.addAll(command, args);
+        Assert.assertArrayEquals(expectedCommand, plugin.getTestCommand());
+    }
+
+    @Test
+    public void testGetAvailablePointsCommand() {
+        String[] command = new String[]{"Rscript"};
+        String[] args;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            args = new String[]{"-e", "\"library('tmcRtestrunner');run_available_points()\""};
+        } else {
+            args = new String[]{"-e", "library(tmcRtestrunner);run_available_points()"};
+        }
+        String[] expectedCommand = ArrayUtils.addAll(command, args);
+        Assert.assertArrayEquals(expectedCommand, plugin.getAvailablePointsCommand());
+    }
+
+    @Test
+    public void testGetPluginName() {
+        assertEquals("r", plugin.getLanguageName());
+    }
+
+    @Test
+    public void testScanExercise() {
+        plugin.scanExercise(simpleAllTestsPassProject, "main.R");
+        File availablePointsJson = new File(simpleAllTestsPassProject.toAbsolutePath().toString()
+                + "/.available_points.json");
+
+        assertTrue(availablePointsJson.exists());
+    }
+
+    @Test
+    public void testScanExerciseInTheWrongPlace() {
+        plugin.scanExercise(simpleAllTestsPassProject, "ar.R");
+        Path availablePointsJson = simpleAllTestsPassProject.resolve(".available_points.json");
+        ImmutableList<TestDesc> re = null;
+        try {
+            re = new RExerciseDescParser(availablePointsJson).parse();
+        } catch (IOException e) {
```

Maybe just throw this exception from the test. It would certainly make this method a bit cleaner.